src/drivers/cmakeDriver.ts

```diff
@@ -1730,6 +1747,67 @@ export abstract class CMakeDriver implements vscode.Disposable {
         }
     }
 
+    private async _showTaskQuickPick(tasks: vscode.Task[]): Promise<vscode.Task | undefined> {
+        interface TaskItem extends vscode.QuickPickItem {
+            task: vscode.Task;
+        }
+        const choices = tasks.map((t): TaskItem => ({ label: t.name, task: t }));
+        const sel = await vscode.window.showQuickPick(choices, { placeHolder: localize('select.build.task', 'Select build task') });
+        return sel ? sel.task : undefined;
+    }
+
+    private async _findBuildTask(targets?: string[]): Promise<vscode.Task | undefined> {
+
+        // Fetch all CMake task
+        const tasks = (await vscode.tasks.fetchTasks({ type: 'cmake' })).filter(task => ((task.group?.id === vscode.TaskGroup.Build.id)));
+
+        // There is only one found - stop searching
+        if (tasks.length === 1) {
+            return tasks[0];
+        }
+
+        // Get only tasks which target matches to one which is requested
+        const targetTasks = tasks.filter(task =>
+            (task.definition as CMakeTaskDefinition).targets === targets);
+
+        if (targetTasks.length > 0) {
+            // Only one found - just return it
+            if (targetTasks.length === 1) {
+                return targetTasks[0];
+            } else {
+                // More than one - check if there is one marked as default
+                const defaultTargetTask = targetTasks.filter(task => task.group?.isDefault);
+                if (defaultTargetTask.length === 1) {
+                    return defaultTargetTask[0];
+                }
+                // None or more than one mark as default - show quick picker
+                return this._showTaskQuickPick(targetTasks);
+            }
+        }
+
+        // Get only tasks with no targets set (template tasks)
```

Comment:
What does "template task" mean in this context?  Is the vscode API returning our unconfigured tasks too?

Reply:
Any task with type cname which does not have target(s) defined.